diff --git a/app/src/main/jni/librime_jni/rime_jni.cc b/app/src/main/jni/librime_jni/rime_jni.cc
index 01c10d02..0416914d 100644
--- a/app/src/main/jni/librime_jni/rime_jni.cc
+++ b/app/src/main/jni/librime_jni/rime_jni.cc
@@ -17,7 +17,7 @@ extern void rime_require_module_charcode();
 extern void rime_require_module_octagram();
 extern void rime_require_module_predict();
 // librime is compiled as a static library, we have to link modules explicitly
-static void declare_librime_module_dependencies() {
+extern "C" void declare_librime_module_dependencies() {
   rime_require_module_lua();
   rime_require_module_charcode();
   rime_require_module_octagram();
@@ -173,12 +173,6 @@ static inline bool is_rime_running() { return Rime::Instance().isRunning(); }
 
 GlobalRefSingleton *GlobalRef;
 
-JNIEXPORT jint JNICALL JNI_OnLoad(JavaVM *jvm, void *reserved) {
-  GlobalRef = new GlobalRefSingleton(jvm);
-  declare_librime_module_dependencies();
-  return JNI_VERSION_1_6;
-}
-
 extern "C" JNIEXPORT void JNICALL Java_com_osfans_trime_core_Rime_startupRime(
     JNIEnv *env, jclass clazz, jstring shared_dir, jstring user_dir,
     jboolean full_check) {

