diff --git a/app/src/main/jni/librime_jni/rime_jni.cc b/app/src/main/jni/librime_jni/rime_jni.cc
index 0dfd132b..419d3a07 100644
--- a/app/src/main/jni/librime_jni/rime_jni.cc
+++ b/app/src/main/jni/librime_jni/rime_jni.cc
@@ -11,7 +11,7 @@ extern void rime_require_module_lua();
 extern void rime_require_module_charcode();
 extern void rime_require_module_octagram();
 // librime is compiled as a static library, we have to link modules explicitly
-static void declare_librime_module_dependencies() {
+extern "C" void declare_librime_module_dependencies() {
   rime_require_module_lua();
   rime_require_module_charcode();
   rime_require_module_octagram();
@@ -132,11 +132,11 @@ static inline bool is_rime_running() { return Rime::Instance().isRunning(); }

 GlobalRefSingleton *GlobalRef;

-JNIEXPORT jint JNICALL JNI_OnLoad(JavaVM *jvm, void *reserved) {
+/*JNIEXPORT jint JNICALL JNI_OnLoad(JavaVM *jvm, void *reserved) {
   GlobalRef = new GlobalRefSingleton(jvm);
   declare_librime_module_dependencies();
   return JNI_VERSION_1_6;
-}
+}*/

 extern "C" JNIEXPORT void JNICALL Java_com_osfans_trime_core_Rime_startupRime(
     JNIEnv *env, jclass clazz, jstring shared_dir, jstring user_dir,
